sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -111,3 +111,49 @@ def update(self, signature):
         if not isinstance(signature, six.string_types):
             raise TypeError("The signature used for updating must be a string.")
         self._signature = signature
+
+
+class AzureNamedKeyCredential(object):
+    """Credential type used for working with any service needing a named key that follows patterns
+    established by the other credential types.
+
+    :param str name: The name of the credential used to authenticate to an Azure service.
+    :param str key: The key used to authenticate to an Azure service.
+    :raises: TypeError
+    """
+    def __init__(self, name, key):
+        # type: (str, str) -> None
+        if not isinstance(name, six.string_types) or not isinstance(key, six.string_types):
+            raise TypeError("Both name and key must be Strings.")
+        self._name = name
+        self._key = key
+
+    @property
+    def name(self):
+        # type () -> str
+        """The value of the configured name.
+
+        :rtype: str
+        """
+        return self._name
+
+    @property
+    def key(self):
+        # type () -> str
+        """The value of the configured key.
+
+        :rtype: str
+        """
+        return self._key
+
+    def update(self, name, key):
+        # type: (str, str) -> None
+        """Update the named key credential.
+
+        Both name and key must be provided in order to update the named key credential.
+        Individual attributes cannot be updated.
+        """
+        if not isinstance(name, six.string_types) or not isinstance(key, six.string_types):
+            raise TypeError("Both name and key must be Strings.")
+        self._name = name
```

Comment:
The reason for a single update method rather than two read/write properties is (per my understanding) to make the assignment atomic. This is not an atomic assignment. Is there a specific reason we have an non-atomic `update` method?

Reply:
How common user creates an AzureNamedKeyCredential then modify the name?

It seems to me maybe

```py
 def update(self, key, **kwarg):
```
makes more sense?

@johanste

Reply:
the `update` method is to 
1. be consistent with other creds in python (aka AzureKeyCredential)
2. be consistent with other languages when it comes to `AzureNamedKeyCredential`

Reply:
No. The reason for the update method is to be atomic. You cannot do that unless you expect callers to lock the instance when reading values.